packages/data/addon/services/navi-facts.ts

```diff
@@ -67,7 +67,9 @@ export default class NaviFactsService extends Service {
    * @returns {String} - url for the request
    */
   getDownloadURL(request: RequestV2, options: RequestOptions) {
-    const type = config.navi.dataSources[0].type;
+    const type = config.navi.dataSources.filter(dataSources => {
+      return dataSources.name === request.dataSource;
+    })[0].type;
```

Comment:
Something like this
https://github.com/yahoo/navi/blob/641a570c54f8b9990094b32442bd42b301578816/packages/data/addon/services/navi-dimension.ts#L9
https://github.com/yahoo/navi/blob/641a570c54f8b9990094b32442bd42b301578816/packages/data/addon/services/navi-dimension.ts#L42-L43